modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/QueryTaskExecutorImpl.java

```diff
@@ -74,7 +75,20 @@ public void failureProcessor(FailureProcessor failureProcessor) {
 
     /** {@inheritDoc} */
     @Override public void execute(UUID qryId, long fragmentId, Runnable qryTask) {
-        stripedThreadPoolExecutor.execute(qryTask, hash(qryId, fragmentId));
+        stripedThreadPoolExecutor.execute(
+            () -> {
+                try {
+                    qryTask.run();
+                }
+                catch (Throwable e) {
+                    if (X.cause(e, InterruptedException.class) != null)
+                        Thread.currentThread().interrupt();
+
```

Comment:
```suggestion
                   /*
                   * No exceptions are rethrown here to preserve the current thread from being destroyed,
                   * because other queries may be pinned to the current thread id.
                   * However, unrecoverable errors must be processed by FailureHandler.
                   */
```